api/v1/tigerastatus_types.go

```diff
@@ -104,6 +104,27 @@ type TigeraStatusList struct {
 	Items           []TigeraStatus `json:"items"`
 }
 
+// TigeraStatusReason represents the reason for a particular condition. A reason may be one of the below.
+type TigeraStatusReason string
+
+const (
+	AllObjectsAvailable       TigeraStatusReason = "AllObjectsAvailable"
+	ResourceNotReady          TigeraStatusReason = "ResourceNotReady"
+	PodFailure                TigeraStatusReason = "PodFailure"
+	CertificateError          TigeraStatusReason = "CertificateError"
+	InvalidConfigurationError TigeraStatusReason = "InvalidConfigurationError"
+	ResourceCreateError       TigeraStatusReason = "ResourceCreateError"
+	ResourceMigrationError    TigeraStatusReason = "ResourceMigrationError"
+	ResourceNotFound          TigeraStatusReason = "ResourceNotFound"
+	ResourcePatchError        TigeraStatusReason = "ResourcePatchError"
+	ResourceReadError         TigeraStatusReason = "ResourceReadError"
+	ResourceScalingError      TigeraStatusReason = "ResourceScalingError"
```

Comment:
I think in essence scaling is just modifying an integer in a resource and update it. Therefore, I think we can remove this and use ResourcePatchError/ResourceUpdateError